### PR TITLE
8225313: serviceability/jvmti/HeapMonitor/MyPackage/HeapMonitorStatObjectCorrectnessTest.java failed with Unexpected high difference percentage

### PR DESCRIPTION
**Pull request by lmesnik**
Fixed as suggested (increasing accepted error rate and a number of iteration). Verified by 2000 runs in our CI lab.The average test execution time ~5s. The bugs in the problemlist are duplicates of this issue.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8225313](https://bugs.openjdk.java.net/browse/JDK-8225313): serviceability/jvmti/HeapMonitor/MyPackage/HeapMonitorStatObjectCorrectnessTest.java failed with Unexpected high difference percentage

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Kevin Walls](https://openjdk.java.net/census#kevinw) (@kevinjwalls - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4864/head:pull/4864` \
`$ git checkout pull/4864`

Update a local copy of the PR: \
`$ git checkout pull/4864` \
`$ git pull https://git.openjdk.java.net/jdk pull/4864/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4864`

View PR using the GUI difftool: \
`$ git pr show -t 4864`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4864.diff">https://git.openjdk.java.net/jdk/pull/4864.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/serviceability/jvmti/HeapMonitor/MyPackage/HeapMonitorStatObjectCorrectnessTest.java [hunk 1]
```diff
@@ -36,11 +36,15 @@
 /** This test is checking the object allocation path works with heap sampling. */
 public class HeapMonitorStatObjectCorrectnessTest {
 
-  // Do 200000 iterations and expect maxIteration / multiplier samples.
-  private static final int maxIteration = 200000;
+  // Do 400000 iterations and expect maxIteration / multiplier samples.
+  private static final int maxIteration = 400_000;
   private static BigObject obj;
 
-  private native static boolean statsHaveExpectedNumberSamples(int expected, int percentError);
```
Comment (lmesnik) [inline test/hotspot/jtreg/serviceability/jvmti/HeapMonitor/MyPackage/HeapMonitorStatObjectCorrectnessTest.java:43]: The method is not implemented and not used.

**Review by dholmes-ora — APPROVED** [review]
Hi Leonid,

These changes seem quite reasonable.

There is a comment in HeapMonitorStatIntervalTest.java that still refers to a 10% threshhold.

Thanks,
David

**Review by dholmes-ora — APPROVED** [review]
Update looks good.

Thanks.

**Comment by lmesnik** [comment]
testing shows that HeapMonitorStatIntervalTest.java needed to make more iterations a well as other fixed tests.

**Review by dholmes-ora — APPROVED** [review]

**Review by kevinjwalls — APPROVED** [review]

**Pull request closed without merging**